Add test cases and examples of App Configuration. (#6)
isra-fel/azure-powershell: src/AppConfiguration/examples/Update-AzAppConfigurationStore.md

> ### Example 1: Update encryption key of an app conifguration store that IdentityType is "SystemAssigned"

Reviewer: ```suggestion
### Example 1: Enable data encryption of the app conifguration store by system-assigned managed identity
```

```powershell
PS C:\> $key = Add-AzKeyVaultKey -VaultName kv-Name -Name key-Name -Destination 'Software'
PS C:\> $systemAssignedAppStore = New-AzAppConfigurationStore -Name appconfig-test11 -ResourceGroupName azpwsh-manual-test -Location $env.location -Sku 'standard' -IdentityType "SystemAssigned"
PS C:\> Set-AzKeyVaultAccessPolicy -VaultName kv-Name -ObjectId $systemAssignedAppStore.IdentityPrincipalId -PermissionsToKeys get,unwrapKey,wrapKey -PassThru
PS C:\> Update-AzAppConfigurationStore -Name appconfig-test11 -ResourceGroupName azpwsh-manual-test -EncryptionKeyIdentifier $key.Id

Location Name             Type
-------- ----             ----
eastus   appconfig-test01 Microsoft.AppConfiguration/configurationStores
```

> This command updates encryption key of an app conifguration store that IdentityType is "UserAssigned".

Reviewer: ```suggestion
This command enables data encryption by a key stored in Azure Key Vault using a system-assigned managed identity.
The vault must have enabled soft-delete and purge-protection, and the managed identity must have these key permissions: get, wrapKey, unwrapKey.
```

> ### Example 2: Update encryption key of an app conifguration store that IdentityType is "UserAssigned" 

Reviewer: ```suggestion
### Example 2: Enable data encryption of the app conifguration store by user-assigned managed identity
```

```powershell
PS C:\> $key = Add-AzKeyVaultKey -VaultName kv-Name -Name key-Name -Destination 'Software'
PS C:\> $assignedIdentity = New-AzUserAssignedIdentity -ResourceGroupName azpwsh-manual-test -Name assignedIdentity
PS C:\> New-AzAppConfigurationStore -Name appconfig-test11 -ResourceGroupName azpwsh-manual-test -Location $env.location -Sku 'standard' -IdentityType "UserAssigned"
PS C:\> Set-AzKeyVaultAccessPolicy -VaultName kv-Name -ObjectId $assignedIdentity.PrincipalId -PermissionsToKeys get,unwrapKey,wrapKey -PassThru
PS C:\> Update-AzAppConfigurationStore -ResourceGroupName azpwsh-manual-test -Name appconfig-test11 -EncryptionKeyIdentifier $key.Id -KeyVaultIdentityClientId $assignedIdentity.ClientId

Location Name             Type
-------- ----             ----
eastus   appconfig-test10 Microsoft.AppConfiguration/configurationStores
```

> This command updates encryption key of an app conifguration store that IdentityType is "SystemAssigned".

Reviewer: ```suggestion
This command enables data encryption by a key stored in Azure Key Vault using a user-assigned managed identity.
The user-assigned identity must have been set with `-UserAssignedIdentity`.
The vault must have enabled soft-delete and purge-protection, and the managed identity must have these key permissions: get, wrapKey, unwrapKey.
```

> ### Example 3: Update encryption key to null of an app conifguration store by pipeline.

Reviewer: ```suggestion
### Example 3: Disable encryption
```

```powershell

> PS C:\> $appConf = Get-AzAppConfigurationStore -ResourceGroupName azpwsh-manual-test -Name appconfig-test10 | Update-AzAppConfigurationStore -EncryptionKeyIdentifier $null

Reviewer: ```suggestion
PS C:\> Get-AzAppConfigurationStore -ResourceGroupName azpwsh-manual-test -Name appconfig-test10 | Update-AzAppConfigurationStore -EncryptionKeyIdentifier $null
```

Author: Done

Location Name             Type
-------- ----             ----
eastus   appconfig-test10 Microsoft.AppConfiguration/configurationStores
```

> This command updates encryption key to null of an app conifguration store.

Reviewer: ```suggestion
To disable encryption, set `EncryptionKeyIdentifier` to `$null`.
```

Author: Done

> ### Example 3: Update sku and tag of an app conifguration store by pipeline.

Reviewer: ```suggestion
### Example 4: Update sku and tag of an app conifguration store by pipeline.
```

```powershell

> PS C:\> $appConf = Get-AzAppConfigurationStore -ResourceGroupName azpwsh-manual-test -Name appconfig-test10 | Update-AzAppConfigurationStore -Sku 'standard' -Tag @{'key'='update'}

Reviewer: ```suggestion
PS C:\> Get-AzAppConfigurationStore -ResourceGroupName azpwsh-manual-test -Name appconfig-test10 | Update-AzAppConfigurationStore -Sku 'standard' -Tag @{'key'='update'}
```

Author: Done
